.github/workflows/test.yml

```
@@ -271,6 +280,47 @@ jobs:
           command: check
           args: --workspace --all-targets
 
+  build_turborepo:
+    name: Build Turborepo
+    needs: [determine_jobs, rust_prepare]
+    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true' || needs.determine_jobs.outputs.node == 'true'
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

```suggestion
    needs: [determine_jobs, rust_prepare]
    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true'
    strategy:
      matrix:
        os:
          - name: ubuntu
            runner: ubuntu-latest
          - name: macos
            runner: macos-latest
          - name: windows
            runner: windows-latest
    runs-on: ${{ matrix.os.runner }}
    name: Build Turborepo for ${{ matrix.os.name }}
```

Seems like `runs-on` is missing here?

You could switch to `runner: ubuntu-latest-16-core-oss` if you want this to be faster, but it costs ~$3/h